alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultCustomCertificateAccessor.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * alert-database
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.database.api;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.persistence.accessor.CustomCertificateAccessor;
+import com.synopsys.integration.alert.common.persistence.model.CustomCertificateModel;
+import com.synopsys.integration.alert.database.certificates.CustomCertificateEntity;
+import com.synopsys.integration.alert.database.certificates.CustomCertificateRepository;
+
+@Component
+public class DefaultCustomCertificateAccessor implements CustomCertificateAccessor {
+    private CustomCertificateRepository customCertificateRepository;
+
+    @Autowired
+    public DefaultCustomCertificateAccessor(CustomCertificateRepository customCertificateRepository) {
+        this.customCertificateRepository = customCertificateRepository;
+    }
+
+    @Override
+    public CustomCertificateModel storeCertificate(CustomCertificateModel certificateModel) throws AlertDatabaseConstraintException {
+        if (null == certificateModel) {
+            throw new AlertDatabaseConstraintException("The certificate model cannot be null");
+        }
+
+        String alias = certificateModel.getAlias();
+        if (StringUtils.isBlank(alias)) {
+            throw new AlertDatabaseConstraintException("The field 'alias' cannot be blank");
+        }
+
+        String certificateContent = certificateModel.getCertificateContent();
+        if (StringUtils.isBlank(certificateContent)) {
+            throw new AlertDatabaseConstraintException("The field 'certificateContent' cannot be blank");
+        }
+
+        CustomCertificateEntity entityToSave = new CustomCertificateEntity(alias, certificateContent);
```

Comment:
I'm not sure that we want to store the certificateContent in our DB. What would the use be? It should just be stored in the Java truststore

Reply:
Currently the certificates get regenerated each time the container is started.  So we would need to have a startup component to import the certificates stored in the database into the truststore.  We could look into saving the certificate into the volume but if the cfssl container gets updated and generates new certificates with different attributes then the Alert container may not be able to communicate with the other containers in a deployment with Black Duck.